frontend/src/components/pages/search/Search.tsx

```diff
@@ -1,10 +1,123 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Layout from '../../shared/layout/Layout';
+import UserList from '../../shared/userList/UserList';
+import Users from '../../mocks/users-mock.json';
+import { Box } from '@material-ui/core';
+import { CountryDropdown } from 'react-country-region-selector';
+// import { countries } from "typed-countries";
+import './search.scss';
 
 const Search: React.FC = () => {
+  const [name, setName] = useState('');
+  const [country, setCountry] = useState('');
+  const [checkbox, setCheckbox] = useState({
+    male: false,
+    female: false,
+    online: false
+  });
+
+  const [selects, setSelects] = useState({
+    lowAge: '13',
+    highAge: '99',
+    language: '',
+    level: ''
+  });
+
+  const selectCountry = (val: string) => {
+    setCountry(val);
+  };
+  const handleName = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+  };
+
+  const handleCheckbox = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setCheckbox({ ...checkbox, [e.target.name]: e.target.checked });
+  };
+
+  const handleSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelects({ ...selects, [e.target.name]: e.target.value });
+  };
+
+  const getGender = () => {
+    if ((!checkbox.female && !checkbox.male) || (checkbox.female && checkbox.male)) {
+      return '';
+    }
+    return Boolean(checkbox.male) ? 'male' : 'female';
+  };
+
+  const getOptions = (arr: (string | number)[]) => {
+    return arr.map((item, i) => (
+      <option key={`${item}${i}`} value={item}>
+        {item}
+      </option>
+    ));
+  };
+
+  const findUsers = (e: React.FormEvent) => {
+    const request = {
+      name: name,
+      lowAge: selects.lowAge,
+      highAge: selects.highAge,
+      sex: getGender(),
+      country: country,
+      language: selects.language,
+      level: selects.level,
+      online: checkbox.online
+    };
+    e.preventDefault();
+    console.log(JSON.stringify(request));
+  };
+
+  const years: number[] = [];
+  for (let i = 13; i < 100; i++) {
+    years.push(i);
+  }
```

Comment:
use constants for limits

also maybe 
`const years = [...Array(hightLimit - lowLimit)].map((_, index) => index+lowLimit)` ? 
What do you think?